src/rowQueue.ts

```diff
@@ -0,0 +1,228 @@
+      this.insert();
+    }
+    this.batch.add(row, callback);
+
+    if (this.batch.isFull()) {
+      this.insert();
+    } else if (!this.pending) {
+      const {maxMilliseconds} = this.batchOptions!;
+      this.pending = setTimeout(() => {
+        this.insert();
+      }, maxMilliseconds);
+    }
+  }
+  /**
+   * Cancels any pending inserts and calls _insert immediately.
+   */
+  insert(callback?: InsertRowsCallback): void {
+    const {rows, callbacks} = this.batch;
+
+    this.batch = new RowBatch(this.batchOptions!);
+
+    if (this.pending) {
+      clearTimeout(this.pending);
+      delete this.pending;
+    }
+    if (rows.length > 0) {
+      this._insert(rows, callbacks, callback);
+    }
+  }
+
+  /**
+   * Accepts a batch of rows and inserts them into table.
+   *
+   * @param {object[]} rows The rows to insert.
+   * @param {InsertCallback[]} callbacks The corresponding callback functions.
+   * @param {function} [callback] Callback to be fired when insert is done.
+   */
+  _insert(
+    rows: RowMetadata | RowMetadata[],
+    callbacks: InsertRowsCallback[],
+    cb?: InsertRowsCallback
+  ): void {
+    const json = extend(true, {}, this.insertRowsOptions, {rows});
+
+    delete json.createInsertId;
+    delete json.partialRetries;
+    delete json.raw;
+
+    this.table.request(
+      {
+        method: 'POST',
+        uri: '/insertAll',
+        json,
+      },
+      (err, resp) => {
+        const partialFailures = (resp.insertErrors || []).map(
+          (insertError: GoogleErrorBody) => {
+            return {
+              errors: insertError.errors!.map(error => {
+                return {
+                  message: error.message,
+                  reason: error.reason,
+                };
+              }),
+              // eslint-disable-next-line @typescript-eslint/no-explicit-any
+              row: rows[(insertError as any).index],
+            };
+          }
+        );
+
+        if (partialFailures.length > 0) {
+          err = new common.util.PartialFailureError({
+            errors: partialFailures,
+            response: resp,
+          } as GoogleErrorBody);
+
+          callbacks.forEach(callback => callback!(err, resp));
+          this.stream.emit('error', err);
+        } else {
+          callbacks.forEach(callback => callback!(err, resp));
+          this.stream.emit('response', resp);
+          cb!(err, resp);
+        }
+        cb!(err, resp);
+      }
+    );
+  }
+
+  /**
+   * Sets the batching options.
+   *
+   *
+   * @param {RowBatchOptions} [options] The batching options.
+   */
+  setOptions(options?: RowBatchOptions): void {
+    const defaults = {
+      maxBytes: defaultOptions.maxOutstandingBytes,
+      maxRows: defaultOptions.maxOutstandingRows,
+      maxMilliseconds: defaultOptions.maxDelayMillis,
+    };
+
+    // TODO: set individual defaults for any unset options
+    // const opts = typeof options === 'object' ? options : defaults;
```

Comment:
Is this meant to be uncommented or removed?

Reply:
Ah, meant to be updated per Megan's review. Done now!